### ISSUE TITLE
respect k_image_data_format() everywhere?

### ISSUE THREAD

**Issue by t-kalinowski** [rstudio/keras#662]
Hello, I'm not sure if this is intentional or not, but the `data_format` argument is hard coded in a handful of functions as `"channels_last"`. Ideally, I think the default arg would be `k_image_data_format()` (or maybe `NULL`?). If the hard-coded `"channels_last"` default is intentional, then the documentation should be updated for the `data_format` arg throughout, since the current documentation states that the value in `~/.keras/keras.json` is used, when it is not. 

Some functions where I've encountered this are 
```R
layer_conv_1d
layer_separable_conv_1d
layer_average_pooling
```

**Comment by jjallaire** [comment]
I think you are correct that it should probably be `NULL` (assuming that will result in the underlying library call resolving to `k_image_data_format()`

**Comment by t-kalinowski** [comment]
Well, unfortunately I don't have time to do a proper pull request, but I figured I'd share my temporary band-aid fix that I ended up putting in my project's `.Rprofile`

```R
  setHook(packageEvent("keras", "onLoad"),
          function(pkgname, pkgpath) {
            ns <- getNamespace("keras")
            nms <- ls(ns)
            
            for (nm in nms) {
              fun <- getFromNamespace(nm, ns)
              if(!is.function(fun)) next
              fml <- formals(fun)
              if ("data_format" %in% names(fml)) {
                formals(fun)$data_format <- quote(k_image_data_format())
                assignInNamespace(nm, fun, "keras")
              }
            }
            
            assignInNamespace("k_is_tensor",
                              function(x)
                                inherits(x, "tensorflow.tensor"), "keras")
          })
```

Making all of keras respect the set data_format is actually is actually going to be a little more tricky. In the python keras implementation, there a bunch of defaults that also assume everything is channels_last (e.g., the `axis` argument in softmax activation, batch_normalization), and in a bunch of places on the R side I've had to sprinkle arguments like this:
```R 
axis = switch(k_image_data_format(), channels_first = 0L, channels_last = -1L)
```
(and also, lose the convenience of being able to just pass a character string like `"softmax"`)

**Comment by skeydan** [comment]
Hi, thanks for describing the workaround (and the analysis!). As it's not a matter of a quick fix if we want to do it "right" (if the Python version itself is not consistent, as you say), I'll close this but tag it "howto" for easy retrieval, if that's okay with you.
